code/datums/wires/particle_accelerator.dm

```diff
@@ -1,67 +1,52 @@
 /datum/wires/particle_acc/control_box
 	wire_count = 5
 	holder_type = /obj/machinery/particle_accelerator/control_box
+	proper_name = "Particle accelerator control"
+	window_x = 361
+	window_y = 22
 
-#define PARTICLE_TOGGLE_WIRE 1 // Toggles whether the PA is on or not.
-#define PARTICLE_STRENGTH_WIRE 2 // Determines the strength of the PA.
-#define PARTICLE_INTERFACE_WIRE 4 // Determines the interface showing up.
-#define PARTICLE_LIMIT_POWER_WIRE 8 // Determines how strong the PA can be.
+/datum/wires/particle_acc/control_box/New(atom/_holder)
+	wires = list(WIRE_PARTICLE_POWER, WIRE_PARTICLE_STRENGTH, WIRE_PARTICLE_INTERFACE, WIRE_PARTICLE_POWER_LIMIT)
+	return ..()
 
-/datum/wires/particle_acc/control_box/GetWireName(index)
-	switch(index)
-		if(PARTICLE_TOGGLE_WIRE)
-			return "Power Toggle"
-		
-		if(PARTICLE_STRENGTH_WIRE)
-			return "Strength"
-		
-		if(PARTICLE_INTERFACE_WIRE)
-			return "Interface"
-			
-		if(PARTICLE_LIMIT_POWER_WIRE)
-			return "Maximum Power"
-
-/datum/wires/particle_acc/control_box/CanUse(mob/living/L)
+/datum/wires/particle_acc/control_box/interactable(mob/user)
 	var/obj/machinery/particle_accelerator/control_box/C = holder
 	if(C.construction_state == 2)
-		return 1
-	return 0
+		return TRUE
+	return FALSE
 
-/datum/wires/particle_acc/control_box/UpdatePulsed(index)
+/datum/wires/particle_acc/control_box/on_pulse(wire)
 	var/obj/machinery/particle_accelerator/control_box/C = holder
-	switch(index)
-
-		if(PARTICLE_TOGGLE_WIRE)
+	switch(wire)
+		if(WIRE_PARTICLE_POWER)
 			C.toggle_power()
 
-		if(PARTICLE_STRENGTH_WIRE)
+		if(WIRE_PARTICLE_STRENGTH)
 			C.add_strength()
 
-		if(PARTICLE_INTERFACE_WIRE)
+		if(WIRE_PARTICLE_INTERFACE)
 			C.interface_control = !C.interface_control
 
-		if(PARTICLE_LIMIT_POWER_WIRE)
+		if(WIRE_PARTICLE_POWER_LIMIT)
 			C.visible_message("[bicon(C)]<b>[C]</b> makes a large whirring noise.")
 	..()
 
-/datum/wires/particle_acc/control_box/UpdateCut(index, mended)
+/datum/wires/particle_acc/control_box/on_cut(wire, mend)
 	var/obj/machinery/particle_accelerator/control_box/C = holder
-	switch(index)
-
-		if(PARTICLE_TOGGLE_WIRE)
-			if(C.active == !mended)
+	switch(wire)
+		if(WIRE_PARTICLE_POWER)
+			if(C.active == !mend)
 				C.toggle_power()
 
-		if(PARTICLE_STRENGTH_WIRE)
-
+		if(WIRE_PARTICLE_STRENGTH)
 			for(var/i = 1; i < 3; i++)
```

Comment:
Would say to use the more efficient loop type here, but then DM would complain about a var being defined but not used. Ah well.

Reply:
I can switch to the other loop with no problems